src/btree/bt_split.c

```diff
@@ -1605,8 +1606,8 @@ __split_multi_inmem_final(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *mul
     WT_UPDATE **tmp;
     uint32_t i, slot;
 
-    /* If we have saved updates, we must have decided to restore them to the new page. */
-    WT_ASSERT(session, multi->supd_entries == 0 || multi->supd_restore);
```

Comment:
We can keep this assert by doing:

WT_ASSERT(session, F_ISSET(btree, WT_BTREE_DISAGGREGATED) || multi->supd_entries == 0 || multi->supd_restore);